src/state/persisted/index.web.ts

```diff
@@ -90,6 +91,17 @@ export async function clearStorage() {
 }
 clearStorage satisfies PersistedApi['clearStorage']
 
+function onStorage() {
+  const next = readFromStorage()
+  if (next === _state) {
+    return
+  }
+  if (next) {
+    _state = next
+    _emitter.emit('update')
```

Comment:
Do we need this emit? Looks like we might be able to remove the `update` event now, since we always emit the more modern event with the key that changed.

Reply:
I think we do. If the `BroadcastChannel` thing arrives too early, it will get ignored. And then by the time we get the `storage` event, we don't know what changed exactly, so we have to deopt. We could try to be smarter about this but I'd prefer to keep it simple and then revisit the assumptions.

Reply:
In other words we don't know the `key` if we use the storage event. So this part needs rethinking